core/src/main/scala/org/broadinstitute/dsde/rawls/monitor/migration/WorkspaceMigrationActor.scala

```diff
@@ -0,0 +1,746 @@
+                                     sourceBucketName: GcsBucketName,
+                                     destGoogleProject: GoogleProjectId,
+                                     destBucketName: GcsBucketName)
+  : MigrateAction[Unit] =
+    for {
+      (storageService, googleProjectToBill) <- MigrateAction.asks { env =>
+        (env.storageService, env.googleProjectToBill)
+      }
+      _ <- MigrateAction.liftIO {
+        for {
+          sourceBucketOpt <- storageService.getBucket(
+            googleProjectToBill, // bill "requester-pays" requests to the migration project
+            sourceBucketName,
+            List(BucketGetOption.userProject(destGoogleProject.value))
+          )
+
+          sourceBucket <- IO(sourceBucketOpt.getOrElse(throw noWorkspaceBucketError(migration, sourceBucketName)))
+
+          // todo: CA-1637 do we need to transfer the storage logs for this workspace? the logs are prefixed
+          // with the ws bucket name, so we COULD do it, but do we HAVE to? it's a csv with the bucket
+          // and the storage_byte_hours in it that is kept for 180 days
+          _ <- storageService.insertBucket(
+            googleProject = GoogleProject(destGoogleProject.value),
+            bucketName = destBucketName,
+            labels = Option(sourceBucket.getLabels).map(_.toMap).getOrElse(Map.empty),
+            bucketPolicyOnlyEnabled = true,
+            logBucket = GcsBucketName(GoogleServicesDAO.getStorageLogsBucketName(GoogleProjectId(destGoogleProject.value))).some,
+            location = Option(sourceBucket.getLocation)
+          ).compile.drain
+        } yield ()
+      }
+    } yield ()
+
+
+  final def startBucketTransferJob(migrationId: Long,
+                                   originBucket: GcsBucketName,
+                                   destBucket: GcsBucketName)
+  : MigrateAction[TransferJob] =
+    for {
+      (storageTransferService, googleProject) <- MigrateAction.asks { env =>
+        (env.storageTransferService, env.googleProjectToBill)
+      }
+      transferJob <- MigrateAction.liftIO {
+        for {
+          jobName <- randomSuffix("transferJobs/terra-workspace-migration-")
+          transferJob <- storageTransferService.createTransferJob(
+            jobName = GoogleStorageTransferService.JobName(jobName),
+            jobDescription = s"""Bucket transfer job from "${originBucket}" to "${destBucket}".""",
+            projectToBill = googleProject,
+            originBucket,
+            destBucket,
+            JobTransferSchedule.Immediately
+          )
+        } yield transferJob
+      }
+
+      _ <- inTransaction { _ =>
+        storageTransferJobs
+          .map(job => (job.jobName, job.migrationId, job.destBucket, job.originBucket))
+          .insert((transferJob.getName, migrationId, destBucket.value, originBucket.value))
+      }
+
+    } yield transferJob
+
+
+  final def updateWorkspaceRecord: MigrateAction[Unit] =
```

Comment:
minor: could we move this to after `deleteTemporaryBucket`?

Reply:
Yes we can!